File: applicants_administration/src/entities/ApplicantList/api/applicationDTO.ts
import { CompetitionTypeDTO } from "../../CompetitionType"
import { StudyModeDTO } from "../../StudyMode"
import {SubjectDTO} from "../../Subject";
import {CityDTO} from "../../City";
import {ProgramDTO} from "../../Program";
import {UniversityDTO} from "../../University";
import {FieldOfStudyDTO} from "../../FieldOfStudy";
import {PeriodDTO} from "../../Period";

export class ApplicationDTO {
    application_id: number;
    city: CityDTO;
    university: UniversityDTO;
    field_of_study: FieldOfStudyDTO;
    program: ProgramDTO;
    study_mode: StudyModeDTO;
    competition_type: CompetitionTypeDTO;
    enrollment_period: PeriodDTO;
    budget_seats: number;
    commercial_seats: number;
    targeted_seats: number;
    subject_list: SubjectDTO[];
    application_link: string | undefined
    last_check_date: Date | undefined

    constructor(id: number, city: CityDTO, university: UniversityDTO, fieldOfStudy: FieldOfStudyDTO, program: ProgramDTO, studyMode: StudyModeDTO,
                competitionType: CompetitionTypeDTO, enrollmentPeriod: PeriodDTO, budgetSeats: number,
                commercialSeats: number, targetedSeats: number, subjectList: SubjectDTO[],
                applicationList: string | undefined, lastCheckDate: Date | undefined) {
        this.application_id = id;
        this.city = city;
        this.university = university;
        this.field_of_study = fieldOfStudy;
        this.program = program;
        this.study_mode = studyMode;
        this.competition_type = competitionType;
        this.enrollment_period = enrollmentPeriod;
        this.budget_seats = budgetSeats;
        this.commercial_seats = commercialSeats;
        this.targeted_seats = targetedSeats;
        this.subject_list = subjectList;
        this.application_link = applicationList;
        this.last_check_date = lastCheckDate;
    }
}

export class ApplicationDTOSet {
    applications: ApplicationDTO[];
    total_count: number;

    constructor(applications: ApplicationDTO[], totalCount: number) {
        this.applications = applications;
        this.total_count = totalCount;
    }
}

File: applications_server/api/dto/application_dto.py
import datetime

from api.dto.city_dto import CityDTO
from api.dto.competition_type_dto import CompetitionTypeDTO
from api.dto.educational_program_dto import EducationalProgramDTO
from api.dto.enrollment_period_dto import EnrollmentPeriodDTO
from api.dto.field_of_study_dto import FieldOfStudyDTO
from api.dto.study_mode_dto import StudyModeDTO
from api.dto.subject_dto import SubjectDTO
from api.dto.university_dto import UniversityDTO
from api.dto.with_config import WithConfig


class ApplicationDTO(WithConfig):
    application_id: int
    university: UniversityDTO
    field_of_study: FieldOfStudyDTO
    program: EducationalProgramDTO
    study_mode: StudyModeDTO
    competition_type: CompetitionTypeDTO
    enrollment_period: EnrollmentPeriodDTO
    budget_seats: int
    commercial_seats: int
    targeted_seats: int
    subject_list: list[SubjectDTO]
    application_link: str | None
    last_check_date: datetime.datetime | None


class ApplicationRowDTO(WithConfig):
    competitive_position: int
    insurance_number: str
    submitted_diploma: bool
    has_special_right: bool
    scores: list[dict]
    extra_score: int
    priority: int | None = None
    has_agreement: bool | None = None


class ApplicationFullDTO(WithConfig):
    application: ApplicationDTO
    application_rows: list[ApplicationRowDTO]


class LastApplicationDTO(WithConfig):
    application: ApplicationDTO
    period_list: list[EnrollmentPeriodDTO]


class LastApplicationFullDTO(WithConfig):
    application: LastApplicationDTO
    application_rows: list[ApplicationRowDTO]


class FilteredApplicationDTO(WithConfig):
    applications: list[ApplicationDTO]
    total_count: int
